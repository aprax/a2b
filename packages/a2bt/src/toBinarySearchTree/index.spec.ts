/* eslint-disable max-len */
import toBst from ".";

const testData = [
  ["1"],
  /*
  ["1", "2"],
  ["1", "2", "3"],
  ["1", "2", "3", "4"],
  ["1", "2", "3", "4", "5"],
  ["a", "d", "z"],
  [
    "a",
    "b",
    "c",
    "d",
    "e",
    "f",
    "g",
    "h",
    "i",
    "j",
    "k",
    "l",
    "m",
    "n",
    "o",
    "p",
    "q",
    "r",
    "s",
    "t",
    "u",
    "v",
    "w",
    "x",
    "y",
    "z",
  ],
  [
    1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,
    22, 23, 24, 25, 26, 27,
  ],
  */
];
for (const nodes of testData) {
  it(`should produce a binary search tree ordered array of ${nodes.toString()}`, () => {
    const actual = toBst(nodes);
    expect(actual).toMatchSnapshot();
  });
}
