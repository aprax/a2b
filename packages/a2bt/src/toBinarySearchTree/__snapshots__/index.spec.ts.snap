// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should produce a binary search tree ordered array of ["1","2","3","4","5"] 1`] = `
Array [
  "3",
  "1",
  "4",
  ,
  "2",
  ,
  "5",
]
`;

exports[`should produce a binary search tree ordered array of ["1","2","3","4"] 1`] = `
Array [
  "2",
  "1",
  "3",
  ,
  ,
  ,
  "4",
]
`;

exports[`should produce a binary search tree ordered array of ["1","2","3"] 1`] = `
Array [
  "2",
  "1",
  "3",
]
`;

exports[`should produce a binary search tree ordered array of ["1","2"] 1`] = `
Array [
  "1",
  ,
  "2",
]
`;

exports[`should produce a binary search tree ordered array of ["1"] 1`] = `
Array [
  "1",
]
`;

exports[`should produce a binary search tree ordered array of ["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"] 1`] = `
Array [
  "m",
  "f",
  "t",
  "c",
  "i",
  "p",
  "w",
  "a",
  "d",
  "g",
  "k",
  "n",
  "r",
  "u",
  "y",
  ,
  "b",
  ,
  "e",
  ,
  "h",
  "j",
  "l",
  ,
  "o",
  "q",
  "s",
  ,
  "v",
  "x",
  "z",
]
`;

exports[`should produce a binary search tree ordered array of ["a","d","z"] 1`] = `
Array [
  "d",
  "a",
  "z",
]
`;

exports[`should produce a binary search tree ordered array of [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27] 1`] = `
Array [
  "14",
  "7",
  "21",
  "3",
  "10",
  "17",
  "24",
  "1",
  "5",
  "8",
  "12",
  "15",
  "19",
  "22",
  "26",
  ,
  "2",
  "4",
  "6",
  ,
  "9",
  "11",
  "13",
  ,
  "16",
  "18",
  "20",
  ,
  "23",
  "25",
  "27",
]
`;
